rest-api/src/users/user/user.service.ts

```
@@ -0,0 +1,130 @@
+        },
+        {
+            id: 'ccb06a0a-6948-496a-9894-4e6c81b00a23',
+            isDeleted: false,
+            login: 'agfgd',
+            password: 'pass',
+            age: 20,
+        },
+        {
+            id: 'cda06f72-85f8-4bbb-9209-94250ea85202',
+            isDeleted: true,
+            login: 'aafgd',
+            password: 'pass',
+            age: 20,
+        },
+    ];
+
+    getAll() {
+        return this.usersBase.filter((user) => {
+            return !user.isDeleted;
+        });
+    }
+
+    getAutoSuggestUsers(loginSubstring: string = '', limit?: number) {
+        if (!limit || limit < 1) {
+            limit = this.usersBase.length;
+        }
+        if (loginSubstring || limit) {
+            return this.usersBase
+                .filter((user) => {
+                    return (
+                        user.login.includes(loginSubstring) && !user.isDeleted
+                    );
+                })
+                .sort((a, b) => {
+                    if (a.login.toLowerCase() < b.login.toLowerCase()) {
+                        return -1;
+                    }
+                    if (a.login.toLowerCase() > b.login.toLowerCase()) {
+                        return 1;
+                    }
+                    return 0;
+                })
+                .slice(0, limit);
+        } else {
+            return this.getAll();
+        }
+    }
+
+    getById(id: string) {
+        return this.usersBase.find((user) => {
+            return user.id === id && !user.isDeleted;
+        });
+    }
+
+    createUser(user: CreateUserDto) {
+        let id: string = v4();
+        this.usersBase.push({ id: id, isDeleted: false, ...user });
+        return { id: id, isDeleted: false, ...user };
+    }
+
+    updateUser(id: string, userData: updateUserDto) {
+        let updUser: User = this.usersBase.find((user) => {
+            return user.id === id && !user.isDeleted;
+        });
+        if (updUser) {
```

here, when the user is not found, you can throw an exception to get a 404 code in the response
if (!updUser || user.isDeleted) {
      throw new NotFoundException('User not found');
    }